smaps/src/main/java/com/google/smaps/FileUpload.java

```diff
@@ -81,36 +100,60 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
+    // Get the current session, which contains user-specific data.
+    HttpSession session = request.getSession();
+
+    // Get the error message currently in this session.
+    String fileErrorMessage = (String) session.getAttribute("fileErrorMessage");
+
     // Transfer the String into JavaScript Object (Json).
     Gson gson = new Gson();
-    String errorJson = gson.toJson(errorMessage);
+    String errorJson = gson.toJson(fileErrorMessage);
+
     // Write Json to index.js.
     response.getWriter().println(errorJson);
   }
 
   /**
-   * Takes in the file's input stream and saves it to the tmp directory, and creates the necessary
-   * data structures and sets the necessary fields.
+   * Creates a new file from fileInputStream and saves it to the tmp directory with a random name,
+   * also creates the necessary data structures and sets them to this session.
    */
-  public void uploadFile(InputStream fileInputStream) throws IOException {
-    // Copies the uploaded file to the server and renames it smaps-upload.txt.
-    File fileToSave = new File("/tmp/smaps-upload.txt");
+  public String uploadFile(HttpSession session, InputStream fileInputStream) throws IOException {
+    // Generate a random filename for this user's file.
+    Random rand = new Random();
+    int randomInt = rand.nextInt(1000000);
+    String filename = "/tmp/smaps-upload" + randomInt + ".txt";
```

Comment:
Can you add a dash after smaps-upload so that it reads as
"/tmp/smaps-upload-1234.txt"

Reply:
Yep